test/overseer/store_test.clj

```diff
@@ -24,6 +24,20 @@
   (->> (tcore/minus now (tcore/millis millis-ago))
        std.time/datetime->unix))
 
+(defn test-transact-graph [store]
+  (testing "idempotent transaction"
+    (let [j1 (test-utils/job {:job/id "foo-id" :job/args {:x 1}})
+          graph1 (api/simple-graph j1)
+
+          j2 (test-utils/job {:job/id "foo-id" :job/args {:x 2}})
+          graph2 (api/simple-graph j2)]
+      (is (= graph1 (core/transact-graph store graph1))
+          "It returns the graph after inserting into the store")
+      (is (= graph2 (core/transact-graph store graph2))
+          "It is idempotent on duplicate job-id insert")
```

Comment:
I think this would be clearer if this was tested on `graph1` rather than `graph2`